subsys/bluetooth/mesh/dfu_slot.h

```diff
@@ -16,48 +16,67 @@
 typedef enum bt_mesh_dfu_iter (*bt_mesh_dfu_slot_cb_t)(
 	const struct bt_mesh_dfu_slot *slot, void *user_data);
 
-/** @brief Register a new DFU image slot for a distributable image.
+/** @brief Get the number of slots committed to the firmware list.
+ *
+ *  @return Number of committed slots.
+ */
+int bt_mesh_dfu_slot_count(void);
+
+/** @brief Reserve a new DFU image slot for a distributable image.
  *
  *  A DFU image slot represents a single distributable DFU image with all its
- *  metadata.
+ *  metadata. The slot data must be set and @c bt_mesh_dfu_slot_commit must be
```

Comment:
Maybe add reference to `bt_mesh_dfu_slot_info_set` and `bt_mesh_dfu_slot_fwid_set` here as they required to be called before committing the slot?